src/APL/vercel-apl.test.ts

```diff
@@ -1,7 +1,33 @@
-import { afterEach, describe, expect, it } from "vitest";
+import { rest } from "msw";
+import { setupServer } from "msw/node";
+import { afterAll, afterEach, beforeAll, describe, expect, it } from "vitest";
 
 import { VercelAPL, VercelAPLVariables } from "./vercel-apl";
 
+export const handlers = [
+  rest.post("https://registerService.example.com/internalError", async (req, res, ctx) =>
+    res(ctx.status(500))
+  ),
+  rest.post("https://registerService.example.com/", async (req, res, ctx) => {
+    const body = await req.json();
+    // Register service will return error when request does not provide token and envs
+    if (!body?.token || !body?.envs) {
+      return res(ctx.status(400));
+    }
+
+    // Check if envs follow the anticipated format
+    const envs = body.envs as Record<string, string | undefined>[];
+    for (const env of envs) {
+      if (typeof env.key === "undefined" || typeof env?.value === "undefined") {
+        return res(ctx.status(400));
+      }
+    }
+    return res(ctx.status(200));
```

Comment:
this body validation is an assumption, but has to be implemented in Appraptor cc @zaiste

Reply:
That's an assumption but wanted to cover the issue I fix with this PR.

Reply:
Good